diff --git a/Makefile b/Makefile
index 8e3d328..c203627 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 DESTDIR?=/
-datadir?=$(DESTDIR)/usr/share
+datadir?=/app/share
 INSTALL=install
 
 SOURCES=$(wildcard *.desktop.in)
@@ -25,7 +25,7 @@ pos:
 
 install: all
 	rm othman-data/quran-kareem.png || :
-	python setup.py install -O2 --root $(DESTDIR)
+	python3 setup.py install --prefix=/app
 	$(INSTALL) -d $(datadir)/applications/
 	$(INSTALL) -m 0644 Othman.desktop $(datadir)/applications/
 	for i in 96 72 64 48 36 32 24 22 16; do \
diff --git a/exe-setup.py b/exe-setup.py
index 9d91139..4dc2a43 100644
--- a/exe-setup.py
+++ b/exe-setup.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/env python3
 import sys, os
 from distutils.core import setup
 from glob import glob
diff --git a/gen-index.py b/gen-index.py
index 761105a..f7ccf15 100755
--- a/gen-index.py
+++ b/gen-index.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/env python3
 # -*- coding: UTF-8 -*-
 
 import sys, os, os.path, time
diff --git a/othman-browser b/othman-browser
index 98f0500..c1b9d2c 100755
--- a/othman-browser
+++ b/othman-browser
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/env python3
 # -*- coding: UTF-8 -*-
 from othman.gtkUi import main
 main()
diff --git a/othman/core.py b/othman/core.py
index e582759..9293356 100644
--- a/othman/core.py
+++ b/othman/core.py
@@ -23,7 +23,7 @@ import sqlite3
 import array
 import threading
 from functools import reduce
-import univaruints
+import othman.univaruints
 
 data_dir = None
 
@@ -203,18 +203,18 @@ class searchIndexer:
         c = cn.cursor()
         c.execute('CREATE TABLE ix (w TEXT PRIMARY KEY NOT NULL, i BLOB)')
         for w in self.d:
-            b = univaruints.incremental_encode(self.d[w].toAyaIdList())
+            b = othman.univaruints.incremental_encode(self.d[w].toAyaIdList())
             self.term_vectors_size += len(b)
             c.execute( 'INSERT INTO ix VALUES(?,?)', (w, sqlite3.Binary(b)) )
         self.terms_count = len(self.d.keys())
         cn.commit()
 
     def _itemFactory(self, r):
-        a = univaruints.incremental_decode(r[1])
+        a = othman.univaruints.incremental_decode(r[1])
         return r[0], searchIndexerItem(a)
 
     def _itemFactory2(self, r):
-        a = univaruints.incremental_decode(r[1])
+        a = othman.univaruints.incremental_decode(r[1])
         return searchIndexerItem(a)
 
     def get(self, w):
diff --git a/othman/gtkUi.py b/othman/gtkUi.py
index 7c33e30..df40416 100644
--- a/othman/gtkUi.py
+++ b/othman/gtkUi.py
@@ -21,7 +21,7 @@ Copyright © 2008-2010, Muayyad Alsadi <alsadi@ojuba.org>
 import sys, os, os.path, time
 import gettext
 from gi.repository import Gtk, Gdk, GLib, Pango, GdkPixbuf
-from core import othmanCore, searchIndexer
+from othman.core import othmanCore, searchIndexer
 
 class searchWindow(Gtk.Window):
     def __init__(self, w):
diff --git a/setup.py b/setup.py
index 39dc675..a4f94e0 100644
--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/env python3
 import sys, os
 from distutils.core import setup
 from glob import glob
